### PR TITLE
SKKU-OSS Go back to main menu & save all scores based on board size.

### PR DESCRIPTION
**Pull request by yjh1534**
Our team merged two branches in one branch.
I'm so sorry to bothering you by of our insufficient experience to open Source project.
We merged two branches,
 1st. Go back to main menu in game playing and selecting board size.
        And also, go back to main menu after view Scoreboard.
 2nd. Save all Scores and statistics in different .txt files based on board size.

Thank you for checking this request~ :)

### CODE REVIEW

**Comment by cawvyoct** [comment]
Will try to review the PR

**Comment by cawvyoct** [comment]
For all PRs submitted, you must run `clang-format`.
This keeps the coding writing-style standardised.
Your PR will not be accepted without running it through `clang-format`.

File: src/game.cpp [hunk 1]
```diff
@@ -321,6 +320,10 @@ void Game::input(KeyInputErrorStatus err) {
     saveState();
     stateSaved = true;
     break;
+  case CODE_RETURN_MENU:
```
Comment (cawvyoct) [inline src/game.cpp]: Good job using an `enum` value. _Magic values_ are not accepted as good coding practice in many projects. 
Enums / Named variables describe values' intentions better to (new) readers of a project.

File: src/game.cpp [hunk 2]
```diff
@@ -493,7 +541,11 @@ ull Game::setBoardSize() {
   constexpr auto no_save_found_text =
       "No saved game found. Starting a new game.";
   constexpr auto board_size_prompt_text =
-      "Enter gameboard size (NOTE: Scores and statistics will be saved only for the 4x4 gameboard): ";
+
+      "Enter gameboard size (3x3 to 10x10). If you want to go back, enter '0': ";
+
+      "Enter gameboard size (3x3 to 10x10). If you want to go back, enter '0' (NOTE: Scores and statistics will be saved only for the 4x4 gameboard): ";
```
Comment (cawvyoct) [inline src/game.cpp]: Code duplication. 
When dealing with merge-conflicts, one must watch carefully for this.
Delete / Edit code which is duplicated in merge-conflicts.

File: src/game.cpp [hunk 3]
```diff
@@ -534,18 +586,27 @@ ull Game::setBoardSize() {
     std::cin.clear();
     std::cin.ignore(std::numeric_limits<std::int32_t>::max(), '\n');
     err = true;
+    if(userInput_PlaySize==0){
+      Menu menu;
+      menu.startMenu();
+    }
+
+	if(userInput_PlaySize == 0){
```
Comment (cawvyoct) [inline src/game.cpp]: Code duplication. Be careful when dealing with merge-conflicts.

File: src/menu.cpp [hunk 4]
```diff
@@ -86,10 +86,43 @@ void Menu::continueGame() {
...
+  std::ostringstream board_size_prompt_richtext;
+
+    constexpr auto choose_playsize = "Enter the size of board (3 to 10). If you want to go back, enter '0': ";
+    const auto invalid_prompt_text ={
+      "Invalid input. Gameboard size should range from ", " to ", "."};
+    error_prompt_richtext << red << std::begin(invalid_prompt_text)[0]
+                            << 3
+                            << std::begin(invalid_prompt_text)[1]
+                            << 10
+                            << std::begin(invalid_prompt_text)[2] << def << "\n\n";
```
Comment (cawvyoct) [inline src/menu.cpp]: Please use `enum` values instead of magic values.

File: src/menu.cpp [hunk 5]
```diff
@@ -86,10 +86,43 @@ void Menu::continueGame() {
...
+    board_size_prompt_richtext << bold_on << " " << choose_playsize<< bold_off;
+
+
+ bool err = false;
+  while(boardsize < 3 || boardsize > 10){
```
Comment (cawvyoct) [inline src/menu.cpp]: Please use `enum` values instead of magic values.

File: src/scores.cpp [hunk 6]
```diff
@@ -86,8 +87,13 @@ void Scoreboard::printScore() {
                     std::to_string(i.score),     i.win ? "Yes" : "No",
                     std::to_string(i.moveCount), std::to_string(i.largestTile),
                     secondsFormat(i.duration)};
+      int hang=0, k;
+      for(k=0;k<data_stats[1].length();k++){
+          if(data_stats[1][k]<0) hang++;
+      }
+      hang = hang/3;
```
Comment (cawvyoct) [inline src/scores.cpp]: I'm confused by what this code is trying to do, could you explain it briefly?

**Comment by tcoyvwac** [comment]
Hi, this PR will close because the PR's code differs from the main codebase a lot. :mag:
If the issue in the project and the solution still exist, please make a new PR using the latest copy of the codebase.
Thanks! :cat2:

**Pull request closed without merging**